tests/test_roadway.py

```diff
@@ -2,31 +2,165 @@
-    node_file = os.path.join(dir,"node.geojson")
+@pytest.mark.roadway
+def test_roadway_read_write(request):
+    print("\n--Starting:",request.node.name)
 
-    print(shape_file)
-    print(link_file)
-    print(node_file)
+    out_path   = "scratch"
+    out_prefix = "t_readwrite"
+    out_dir    = os.path.join(os.getcwd(),out_path)
+    out_shape_file = os.path.join(out_dir,out_prefix+"_"+"shape.geojson")
+    out_link_file  = os.path.join(out_dir,out_prefix+"_"+"link.json")
+    out_node_file  = os.path.join(out_dir,out_prefix+"_"+"node.geojson")
+    time0 = time.time()
+    net = RoadwayNetwork.read(link_file= STPAUL_LINK_FILE, node_file=STPAUL_NODE_FILE, shape_file=STPAUL_SHAPE_FILE, fast=True)
+    time1 = time.time()
+    net.write(filename=out_prefix,path=out_path)
+    time2 = time.time()
+    net_2 = RoadwayNetwork.read(link_file= out_link_file, node_file=out_node_file, shape_file=out_shape_file)
+    time3 = time.time()
 
-    net = RoadwayNetwork.read(link_file= link_file, node_file=node_file, shape_file=shape_file)
-    print(net.nodes_df.columns)
-    net.write(filename="t_readwrite",path='tests')
+    read_time1 = time1-time0
+    read_time2 = time3-time2
+    write_time = time2-time1
 
+    print("TIME, read (w/out valdiation, with): {},{}".format(read_time1, read_time2))
+    print("TIME, write:{}".format(write_time))
+    '''
+    # right now don't have a good way of ignoring differences in rounding
     with open(shape_file, 'r') as s1:
         og_shape = json.loads(s1.read())
         #og_shape.replace('\r', '').replace('\n', '').replace(' ','')
-    with open(os.path.join('tests','t_readwrite_shape.geojson'), 'r')  as s2:
+    with open(os.path.join('scratch','t_readwrite_shape.geojson'), 'r')  as s2:
         new_shape = json.loads(s2.read())
         #new_shape.replace('\r', '').replace('\n', '').replace(' ','')
     assert(og_shape==new_shape)
+    '''
+
+@pytest.mark.roadway
+@pytest.mark.travis
+def test_quick_roadway_read_write(request):
+    print("\n--Starting:",request.node.name)
+
+    out_path   = "scratch"
+    out_prefix = "t_readwrite"
+    out_dir    = os.path.join(os.getcwd(),out_path)
+    out_shape_file = os.path.join(out_dir,out_prefix+"_"+"shape.geojson")
+    out_link_file  = os.path.join(out_dir,out_prefix+"_"+"link.json")
+    out_node_file  = os.path.join(out_dir,out_prefix+"_"+"node.geojson")
+    net = RoadwayNetwork.read(link_file= SMALL_LINK_FILE, node_file=SMALL_NODE_FILE, shape_file=SMALL_SHAPE_FILE, fast=True)
+    net.write(filename=out_prefix,path=out_path)
+    net_2 = RoadwayNetwork.read(link_file= out_link_file, node_file=out_node_file, shape_file=out_shape_file)
+    print("--Finished:",request.node.name)
+
+@pytest.mark.menow
+def test_select_roadway_features(request):
+    print("\n--Starting:",request.node.name)
+    net = RoadwayNetwork.read(link_file= STPAUL_LINK_FILE, node_file=STPAUL_NODE_FILE, shape_file=STPAUL_SHAPE_FILE, fast=True)
+
+    test_selections = { \
```

Comment:
@kulshresthaa 
We should add a bunch of these to satisfy  and make sure they are returning the right links. 
Right now the last one doesn't pass but we should add others and also make sure they are returning the right links.
I will probably set up a notebook to look at this tomorrow if you don't get to it first.

Reply:
Yes, the last one (5) fails because the selection query builder is only looking at the `link` attributes from the facility dict and not doing anything with `A` and `B`. 

In fact, test selection 1-4 are also not using A and B for selection query. 1 and 2 will return same set of links

I should revise the code to include A and B into selection criteria.

Tell me again, how A and B are used. I mean I know the selection has to return links b/w A and B nodes but how to go about it from links_df and nodes_df?

Reply:
> ...how to go about it from links_df and nodes_df

I don't have a brilliant idea, but I would probably investigate using something like OSMNX